### PR TITLE
Use crop hashes to filter out originals from crops

### PR DESCRIPTION
**Pull request by joemcgill**
This is a possible approach to #155, which is a tricky problem since WP core doesn't explicitly describe an image size as a crop vs an original. Feedback would be appreciated.

### CODE REVIEW

File: wp-tevko-responsive-images.php [hunk 1]
```diff
@@ -209,9 +209,19 @@ function tevkori_get_srcset_array( $id, $size = 'thumbnail' ) {
 	// Calculate the image aspect ratio.
 	$img_ratio = $img_height / $img_width;
 
+	// If the image has been cropped, it will contain a unique hash.
+	// We look for that hash and use it later to filter out images that aren't
+	//  of the same crop.
+	$img_cropped = preg_match( '/e\d{13}/', $img_url, $img_crop_hash );
```
Comment (jaspermdegroot) [inline wp-tevko-responsive-images.php]: WP core uses ```'/-e[0-9]{13}\./'``` (wp-admin/includes/image-edit.php) so maybe it's good to use the same regex.

File: wp-tevko-responsive-images.php [hunk 2]
```diff
@@ -209,9 +209,19 @@ function tevkori_get_srcset_array( $id, $size = 'thumbnail' ) {
 	// Calculate the image aspect ratio.
 	$img_ratio = $img_height / $img_width;
 
+	// If the image has been cropped, it will contain a unique hash.
+	// We look for that hash and use it later to filter out images that aren't
+	//  of the same crop.
```
Comment (jaspermdegroot) [inline wp-tevko-responsive-images.php]: Remove extra space at beginning of the line

File: wp-tevko-responsive-images.php [hunk 3]
```diff
@@ -209,9 +209,19 @@ function tevkori_get_srcset_array( $id, $size = 'thumbnail' ) {
...
+	$img_cropped = preg_match( '/e\d{13}/', $img_url, $img_crop_hash );
+
 	// Only use sizes with same aspect ratio.
 	foreach ( $img_sizes as $img ) {
 
+		// filter out images that are from different crops
```
Comment (jaspermdegroot) [inline wp-tevko-responsive-images.php]: Start with capital and end with period.

**Comment by jaspermdegroot** [comment]
The problem as described in https://github.com/ResponsiveImagesCG/wp-tevko-responsive-images/issues/155#issuecomment-135862662 does not only occur when you use the crop tool but also when you (for example) scale and flip an image. The code of this PR handles all cases, but maybe we should change comments and variable names to something more generic ("edited" instead of "cropped")

**Comment by joemcgill** [comment]
+1 good suggestion.

**Comment by joemcgill** [comment]
@jaspermdegroot I updated the logic to filter out non-edits based on your feedback and added tests. I went ahead and merged both since you had already provided feedback, but feel free to look again and let me know if you have any concerns. Thanks.

**Comment by jaspermdegroot** [comment]
@joemcgill - This looks good to me. Only one minor thing; one comment was still about crops (see my comment in the diff).

File: wp-tevko-responsive-images.php [hunk 4]
```diff
@@ -209,10 +209,21 @@ function tevkori_get_srcset_array( $id, $size = 'thumbnail' ) {
...
+	// contain a unique hash. We look for that hash and use it later to filter
+	// out images that are left overs from previous renditions.
+	$img_edited = preg_match( '/-e[0-9]{13}/', $img_url, $img_edit_hash );
+
+	// Loop through available images and only use images that are resized
+	// versions of the same rendition.
 	foreach ( $img_sizes as $img ) {
 
-		// Calculate the new image ratio
+		// Filter out images that are from different crops.
```
Comment (jaspermdegroot) [inline wp-tevko-responsive-images.php]: "Filter out images that are left overs from previous renditions."

**Comment by joemcgill** [comment]
Good catch @jaspermdegroot. Fixed in 5ca463e

**Pull request merged**